File: leetcode/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k.py
# https://leetcode.com/problems/minimum-number-of-operations-to-make-array-xor-equal-to-k/

from typing import List

class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        xor = 0
        for num in nums: xor ^= num
        operations_count = 0
        while xor or k:
            if (xor & 1) ^ (k & 1):
                operations_count += 1
            xor >>= 1
            k >>= 1
        
        return operations_count

File: leetcode/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k.js
// https://leetcode.com/problems/minimum-number-of-operations-to-make-array-xor-equal-to-k/

/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minOperations = function(nums, k) {
  let xor = 0;
  for (const num of nums) xor ^= num;

  let operationsCount = 0;
  while (xor > 0 || k > 0) {
    if ((xor & 1) ^ (k & 1)) operationsCount += 1;
    xor >>= 1;
    k >>= 1;
  }

  return operationsCount;
};
